csharp/.editorconfig

```diff
@@ -0,0 +1,18 @@
+# Remove the line below if you want to inherit .editorconfig settings from higher directories
+root = true
+
+# C# files
+[*.cs]
+
+#### Core EditorConfig Options ####
+
+# Indentation and spacing
+indent_size = 4
+indent_style = space
+tab_width = 4
+
+# New line preferences
+# See https://github.com/protocolbuffers/protobuf/issues/9526
+#end_of_line = lf
```

Comment:
Shouldn't this be uncommented?

Reply:
Not yet - we may well do that as part of fixing #9526, but this PR just adds the editor without trying to address that.